src/metabase/driver/h2.clj

```diff
@@ -195,10 +197,10 @@
 
 (defn- check-read-only-statements [{:keys [database] {:keys [query]} :native}]
   (when query
-    (let [query-classification (classify-query database query)]
-      (when-not (read-only-statements? query-classification)
+    (let [classified-query (classify-query database query)]
```

Comment:
What do you think of changing this name? 

To me, `classified-query` gives the impression that this still contains the query, but it's really just some information about the query.